File: frontend/src/types/note.ts
export type NoteType = 'insight' | 'summary' | 'draft' | 'issue';

export type Note = {
  id: number;
  workflow_id: number;
  title: string;
  content: string;
  note_type: NoteType;
  created_at: string;
  updated_at: string | null;
};

export type NoteCreateInput = {
  workflow_id: number;
  title: string;
  content: string;
  note_type: NoteType;
};

export type NoteUpdateInput = Partial<Omit<NoteCreateInput, 'workflow_id'>>;

File: backend/app/schemas/note.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


NoteType = Literal["insight", "summary", "draft", "issue"]


class NoteCreate(BaseModel):
    workflow_id: int
    title: str = Field(..., max_length=255)
    content: str
    note_type: NoteType = "insight"


class NoteUpdate(BaseModel):
    title: str | None = Field(default=None, max_length=255)
    content: str | None = None
    note_type: NoteType | None = None


class NoteRead(ORMModel):
    id: int
    workflow_id: int
    title: str
    content: str
    note_type: str
    created_at: datetime
    updated_at: datetime | None = None
